### ISSUE TITLE
How is the temporal transformer in Model 2 initialized?

### ISSUE THREAD

**Issue by taoyang1122** [google-research/scenic#351]
Thanks for the great work. I found some details about initialization in Sec 3.4, but I am not sure about the initialization of temporal transformer encoders in Model 2. Are these layers randomly initialized? Thanks.

**Comment by MostafaDehghani** [comment]
@taoyang1122  Thanks for you question!
Can you mention which project you're referring to?

**Comment by taoyang1122** [comment]
Sorry for the confusion. I mean Model 2 in ViViT (Fig. 4). Thanks.

**Comment by MostafaDehghani** [comment]
Yes, in Model 2 (Factorized Encoder), temporal encoder is initialized randomly!

**Comment by taoyang1122** [comment]
Thanks for confirming! Another question is that in Model 2 at the input to the Spatial encoders, are all the input frames sharing the same cls_token or each frame has a sepaprate cls_token? Thanks!

**Comment by MostafaDehghani** [comment]
No problem. 
Yes.  We [create the CLS token inside the `vit_body` function](https://github.com/google-research/scenic/blob/6dd96c5ce73cf2ac3e3e72dd91e558046e0b7136/scenic/projects/vivit/model.py#L554) that we [vmap it](https://github.com/google-research/scenic/blob/6dd96c5ce73cf2ac3e3e72dd91e558046e0b7136/scenic/projects/vivit/model.py#L581), thus we use exactly the same set of parameters (including the cls token) to process different frames.

**Comment by taoyang1122** [comment]
Thanks for your prompt reply. That is really helpful!
